packages/slate-editor/src/modules/editor-v4/EditorV4.tsx

```diff
@@ -272,6 +273,9 @@ const EditorV4: FunctionComponent<EditorV4Props> = ({
             <Slate
                 editor={editor}
                 onChange={(newValue) => {
+                    if (!newValue.every(isBlockNode)) {
```

Comment:
Calling a detailed type-guard on `change` event may turn out to cause performance issues.

Reply:
> Calling a detailed type-guard on change event may turn out to cause performance issues.

In the `slate` repo they have optimized a similar use-case with results cached to a `WeakMap`:

https://github.com/ianstormtaylor/slate/blob/4945a1a27505f59805bbbb630d8e22e47b1f29e5/packages/slate/src/interfaces/node.ts#L388-L399